### PR TITLE
fix: Cannot terminate sidecar when program exits abnormally

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

Cannot terminate sidecar when program exits abnormally

## Issue reference

[issues-507](https://github.com/masastack/MASA.Framework/issues/507)

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptions.cs [hunk 1]
```diff
@@ -9,14 +9,14 @@ namespace Masa.BuildingBlocks.Development.DaprStarter;
...
-public class DaprOptions
+public class DaprOptions : DaprOptionsBase
 {
     /// <summary>
     /// The id for your application, used for service discovery
     /// </summary>
     public string? AppId { get; set; }
 
-    private string _appIdDelimiter = Constant.DEFAULT_APPID_DELIMITER;
+    private string _appIdDelimiter = DaprStarterConstant.DEFAULT_APPID_DELIMITER;
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptions.cs:20]: Processed

File: src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptions.cs [hunk 2]
```diff
@@ -58,76 +58,31 @@ public string AppIdDelimiter
...
     /// </summary>
-    public int? MaxConcurrency
+    public override int? MaxConcurrency
     {
         get => _maxConcurrency;
         set
         {
             if (value != null)
-                MasaArgumentException.ThrowIfLessThanOrEqual(value.Value, (ushort)0, nameof(MaxConcurrency));
+                MasaArgumentException.ThrowIfLessThan(value.Value, -1, nameof(MaxConcurrency));
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptions.cs:74]: Processed

File: src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptions.cs [hunk 3]
```diff
@@ -58,76 +58,31 @@ public string AppIdDelimiter
...
-    public string? ComponentPath { get; set; }
-
     private ushort? _daprGrpcPort;
 
     /// <summary>
     /// The gRPC port for Dapr to listen on
     /// Must be greater than 0
     /// </summary>
-    public ushort? DaprGrpcPort
+    public override ushort? DaprGrpcPort
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptions.cs:86]: Processed

File: src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptionsBase.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    /// The port your application is listening on
+    /// Required. Must be between 0-65535
+    /// </summary>
+    public ushort? AppPort
+    {
+        get => _appPort;
+        set
+        {
+            if (value != null)
+                MasaArgumentException.ThrowIfLessThanOrEqual(value.Value, (ushort)0, nameof(AppPort));
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Development/Masa.BuildingBlocks.Development.DaprStarter/Options/DaprOptionsBase.cs:23]: Processed

**Review by doddgu — APPROVED** [review]

**Pull request merged**